packages/wallet/src/wallet.ts

```diff
@@ -278,18 +300,19 @@ export class DemexWallet {
     const _signOpts: SignTxOpts = {
       ...signOpts,
       tx: {
-        timeoutHeight: signOpts?.tx?.timeoutHeight ?? timeoutHeight,
         ...signOpts?.tx,
+        timeoutHeight: signOpts?.tx?.timeoutHeight ?? timeoutHeight,
       },
       signer: {
+        ...signOpts?.signer,
         accountNumber: signOpts?.signer?.accountNumber ?? accountNumber,
         sequence: signOpts?.signer?.sequence ?? sequence,
-        ...signOpts?.signer,
       }
     };
 
     const signedTx = await this.getSignedTx(address, messages, signingClient, _signOpts);
-    this.updateAccountSequence(address, sequence + 1);
+
+    this.updateWalletAccount(address, { sequence: _signOpts.signer?.sequence! + 1 });
```

Comment:
should abstract `signOpts?.signer?.sequence ?? sequence` into a variable and use that instead.